cpp/src/arrow/array-test.cc

```diff
@@ -1022,6 +1022,39 @@ TEST_F(TestStringBuilder, TestAppendVector) {
   }
 }
 
+TEST_F(TestStringBuilder, TestAppendCStrings) {
```

Comment:
You should also write a test without a `valid_bytes` argument.

Reply:
Exactly.
I've added.